accord-core/src/main/java/accord/coordinate/CoordinatePreAccept.java

```diff
@@ -250,6 +250,16 @@ void onNewEpoch(Topologies prevTopologies, Timestamp executeAt, List<PreAcceptOk
         // TODO (desired, efficiency): check if we have already have a valid quorum for the future epoch
         //  (noting that nodes may have adopted new ranges, in which case they should be discounted, and quorums may have changed shape)
         node.withEpoch(executeAt.epoch(), () -> {
+            TopologyMismatch mismatch = TopologyMismatch.checkForMismatch(node.topology().globalForEpoch(executeAt.epoch()), txnId, route.homeKey(), txn.keys());
+            if (mismatch != null)
+            {
+                Invalidate.invalidate(node, txnId, route, (outcome, failure) -> {
```

Comment:
This should invoke `Propose.proposeInvalidate` not `Invalidate.invalidate`

Reply:
I think we should also set `initialPreAcceptIsDone=true` else we might yet complete successfully as well as propose invalidation, which would be a serious fault.

Reply:
looking into this now

Reply:
made this change